src/index.js

```diff
@@ -1,29 +1,41 @@
-var using = function (values, func) {
-    if (values instanceof Function) {
-        values = values();
+(function (factory) {
+    'use strict';
+    if (typeof define === 'function' && define.amd) {
```

Comment:
instanceof will not work in this case, because variable "define" exists only when the requirejs is available, there will be an error "define is not defined" when somebody wants to use require() version.